src/material/select/select.scss

```diff
@@ -146,3 +146,23 @@ $mat-select-placeholder-arrow-space: 2 * ($mat-select-arrow-size + $mat-select-a
     display: block;
   }
 }
+
+// Based on the spec a form field has two ways of indicating focus in `outline` mode: making the
+// border thicker and floating the label. The problem is that the border is also thicker when the
+// form field is invalid and we only float the label on a `mat-select` when its panel is open or it
+// has a value, resulting in us not having any focus indication when the select is empty and
+// invalid. For this special case we add an underline to the label to indicate focus by targeting
+// a `mat-form-field` label that matches the following criteria:
+// - Uses the `outline` appearance.
+// - Has a `mat-select` form control.
+// - Isn't floating.
+// - Has a label.
+// - Is invalid.
+// - Is empty.
+.mat-form-field-type-mat-select.mat-form-field-appearance-outline:not(.mat-form-field-should-float) {
+  &.mat-form-field-has-label.mat-form-field-invalid {
+    .mat-form-field-label.mat-form-field-empty {
+      border-bottom: solid 2px;
```

Comment:
I've got some mixed feelings about this since it's not in the spec. The alternative is to float the label when the form field is focused, similarly to how we deal with `input`. The problem is that it'll look weird because we have nothing to show instead of the label where an `input` would usually have a blinking caret. 

For reference, here's what it looks like with the underline:
![2020-01-02_10-01-35](https://user-images.githubusercontent.com/4450522/71657532-f8020a80-2d48-11ea-95cb-af048bae51d1.png)

Reply:
Though it doesn't match up with the published spec, MDC uses a 1px outline for non-focused error state and a 2px outline for focused error state. If we want to fix this, we should probably converge on what they're doing rather than invent our own UI.

Reply:
Wouldn't that cause a lot of screenshot diff failures on our end?

Reply:
Yeah, so given that it might be best to just leave this for now and fix it in the MDC version